rbac/core/kafka.py

```diff
@@ -34,12 +38,25 @@ class RBACProducer:
 
     def get_producer(self):
         """Init method to return fake kafka when flag is set to false."""
+        retries = 1
+        max_retries = settings.KAFKA_AUTH.get("retries")
         if not hasattr(self, "producer"):
             if settings.DEVELOPMENT or settings.MOCK_KAFKA or not settings.KAFKA_ENABLED:
                 self.producer = FakeKafkaProducer()
+                logger.info("Fake Kafka producer initialized in development mode")
             else:
                 if settings.KAFKA_AUTH:
-                    self.producer = KafkaProducer(**settings.KAFKA_AUTH)
+                    try:
+                        self.producer = KafkaProducer(**settings.KAFKA_AUTH)
+                        logger.info("Kafka producer initialized successfully")
+                    except KafkaError as e:
+                        logger.error(f"Kafka error during initialization of Kafka producer: {e}")
+                        retries += 1
+                        if retries > max_retries:
+                            logger.critical(f"Failed to initialize Kafka producer after {retries} attempts")
```

Comment:
I believe `if retries > max_retries:` will be never `True`
because if the KafkaError occurs in the `try` block then `retries` value is increased by 1 (so it is 2) and then  the code execution will go to return `self.producer` but this atribute does not exist so it should lead to `NameError` @astrozzc can you help here? I am not familiar how to run Kafka locally so I cannot check my idea, thank you

Reply:
It will be True because the condition loop is **retries <= max_retries**. So for the last iteration, `retries=max_retries` -> raise exception, `retries += 1` -> `retries > max_retries`.
You are right that the self.producer will not be set in this case, we might still want to raise the exception besides logger.critical

Reply:
@astrozzc the loop was added after my comment